tests/Moq.Tests/UnmatchableMatchersFixture.cs

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) 2007, Clarius Consulting, Manas Technology Solutions, InSTEDD.
+// All rights reserved. Licensed under the BSD 3-Clause License; see License.txt.
+
+using System;
+
+using Xunit;
+
+namespace Moq.Tests
+{
+	public class UnmatchableMatchersFixture
+	{
+		[Fact]
+		public void Matchers_that_are_unmatchable_due_to_implicit_conversion_operator_cause_setup_failure()
+		{
+			var mock = new Mock<IX>();
+			var ex = Assert.Throws<ArgumentException>(() => mock.Setup(x => x.UseDateTimeOffset(It.IsAny<DateTime>())));
```

Comment:
For the sake of these tests' independence from external factors, perhaps they shouldn't rely on external types (even ones from the BCL). It might be wiser to define our own types with a `implicit operator`. This would make the operator directly visible and thus make the problem at hand easier to understand.